### PR TITLE
fix: cursor being leaked outside of canvas

### PR DESCRIPTION
**Pull request by dwelle**
fix https://github.com/excalidraw/excalidraw/issues/3160

- moved cursor from `document` to `canvas`. This way it won't ever affect DOM outside the component, but it'll also fix cases where we forget to update the cursor when mousing over the UI, so it's a win-win.
- moved the cursor updating to helper — not sure if the extra function will meaningfully impact perf in tight loops (such as `pointermove` events), but it doesn't look like it

### CODE REVIEW

File: src/element/resizeElements.ts [hunk 1]
```diff
@@ -105,13 +104,6 @@ export const transformElements = (
       );
     }
 
-    // update cursor
-    // FIXME it is not very nice to have this here
-    document.documentElement.style.cursor = getCursorForResizingElement({
-      element,
-      transformHandleType,
-    });
-
```
Comment (dwelle) [inline src/element/resizeElements.ts:114]: From testing it doesn't seem like we need it, but there may be some edge cases. If so, we'll move it up to `App.tsx`.

File: src/utils.ts [hunk 2]
```diff
@@ -160,15 +160,30 @@ export const removeSelection = () => {
...
+  if (!canvas) {
+    return;
+  }
+  canvas.style.cursor = "";
+};
+
+export const setCursor = (canvas: HTMLCanvasElement | null, cursor: string) => {
+  if (canvas) {
+    canvas.style.cursor = cursor;
+  }
```
Comment (lipis) [inline src/utils.ts:173]:
```suggestion
  canvas?.style.cursor = cursor;
```

File: src/utils.ts [hunk 3]
```diff
@@ -160,15 +160,30 @@ export const removeSelection = () => {
 
 export const distance = (x: number, y: number) => Math.abs(x - y);
 
-export const resetCursor = () => {
-  document.documentElement.style.cursor = "";
+export const resetCursor = (canvas: HTMLCanvasElement | null) => {
+  if (!canvas) {
+    return;
+  }
+  canvas.style.cursor = "";
```
Comment (lipis) [inline src/utils.ts:167]:
```suggestion
  canvas?.style.cursor = "";
```

File: src/utils.ts (hunk 2 above)
Comment (dwelle) [reply src/utils.ts:173]: sadly, that's illegal in JS: it'd try to assign `cursor` to `undefined` in case `canvas` was null

**Review by ad1992 — APPROVED** [review]

**Pull request merged**